### ISSUE TITLE
How to add new walletuser without generating private key

### ISSUE THREAD

**Issue by Tschaul** [miguelfreitas/twister-core#369]
Hi,

I want to implement client side crypto. For that I need a way to create a walletuser without telling the private key to the server and without the server generating a new private key for me. Theoretically I could use the "guest" account, but than the rpc method "getlasthave" would not work if there are multiple users on the same server. Does twisterd have a concept of a walletuser without a private key?

Alternatively It would work with a rpc method similar to "getlasthave" which would take a list of users as input for which it returns the last known k-value.

Is there something already or does it need code changes for twisterd?

Greetings

**Comment by Tschaul** [comment]
Just found out that getlasthave has such a new option... 

```
getlasthave <username> | <groupname> [user1,user2...]
```

 But what does <groupname> stand for?

**Comment by Tschaul** [comment]
The second methods signature for "getlasthave" seems to work fine.

But i'd still like this question answered: Does twisterd support walletusers without a known private key?

One hint for that is the method "listwalletusers" is open in public_server_mode. That doesn't make sense if these are only walletusers with private keys.

**Comment by miguelfreitas** [comment]
Good catch! I didn't even remembered this second signature: yes, you may pass the users list and the "groupname" is actually not used at all.

We don't support walletusers without a key. "listwalletusers" is probably incorrectly flagged to public_server_mode.
